tests/test_matrixsampling.py

```diff
@@ -0,0 +1,281 @@
+"""
+Tests for the various sampling classes
+"""
+from __future__ import print_function, division, absolute_import
+
+from itertools import product
+import numpy as np
+from pytest import raises, approx
+import six
+
+from mitxgraders import (
+    RealVectors,
+    ComplexVectors,
+    RealTensors,
+    ComplexTensors,
+    RealMatrices,
+    ComplexMatrices,
+    IdentityMatrixMultiples,
+    SquareMatrices,
+    OrthogonalMatrices,
+    UnitaryMatrices,
+    MathArray,
+    RealInterval,
+    ComplexRectangle,
+    ConfigError
+)
+from mitxgraders.helpers.calc import within_tolerance
+
+def test_vectors():
+    # Test shape, real/complex, norm, MathArray
+    shapes = tuple(range(2, 5))
+    norms = ([2, 6], [3, 4], [4, 12], [1 - 1e-12, 1 + 1e-12])
+
+    for shape in shapes:
+        for norm in norms:
+            vectors = RealVectors(shape=shape, norm=norm)
+            vec = vectors.gen_sample()
+            assert vec.shape == (shape, )
+            assert norm[0] <= np.linalg.norm(vec) <= norm[1]
+            assert np.array_equal(np.conj(vec), vec)
+            assert isinstance(vec, MathArray)
+
+            vectors = ComplexVectors(shape=shape, norm=norm)
+            vec = vectors.gen_sample()
+            assert vec.shape == (shape, )
+            assert norm[0] <= np.linalg.norm(vec) <= norm[1]
+            assert not np.array_equal(np.conj(vec), vec)
```

Comment:
did you want the `isinstance` check for the complex vectors, too? Also below in tensors/matrices...

Reply:
Huh, surprised I missed that (the first time - subsequent times were definitely copy/paste!).